waiter/src/waiter/process_request.clj

```diff
@@ -384,18 +384,18 @@
     (meters/mark! (metrics/service-meter service-id "response-status-rate" (str status)))
     (counters/inc! (metrics/service-counter service-id "request-counts" "waiting-to-stream"))
     (confirm-live-connection-with-abort)
-    (let [request-abort-callback (abort-http-request-callback-factory response)]
+    (let [modified-response (inspect-for-202-async-request-response
```

Comment:
What do you think of `response'`?

Reply:
Did this move because of promises?  You want `:success-async` to be the value received with `@reservation-status-promise`?

Reply:
Yes.

Reply:
Used `response`'

Reply:
I now avoid passing the `reservation-status-promise` to `inspect-for-202-async-request-response`